packages/nx-monorepo/src/nx-types.ts

```diff
@@ -0,0 +1,186 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+export namespace Nx {
+  /**
+   * Configuration for nx targetDependencies.
+   */
+  export type TargetDependencies = { [target: string]: TargetDependency[] };
+
+  /**
+   * Configuration for project specific targets.
+   */
+  export type ProjectTargets = { [target: string]: ProjectTarget };
+
+  /**
+   * Project Target.
+   */
+  export interface ProjectTarget {
+    /**
+     * List of inputs to hash for cache key, relative to the root of the monorepo.
+     *
+     * note: must start with leading /
+     */
+    readonly inputs?: string[];
+    /**
+     * List of outputs to cache, relative to the root of the monorepo.
+     *
+     * note: must start with leading /
+     */
+    readonly outputs?: string[];
+
+    /**
+     * List of Target Dependencies.
+     */
+    readonly dependsOn?: TargetDependency[];
+  }
+
+  /**
+   * Implicit Dependencies map.
+   */
+  export type ImplicitDependencies = { [pkg: string]: string[] };
+
+  /**
+   * Supported enums for a TargetDependency.
+   */
+  export enum TargetDependencyProject {
+    /**
+     * Only rely on the package where the target is called.
+     *
+     * This is usually done for test like targets where you only want to run unit
+     * tests on the target packages without testing all dependent packages.
+     */
+    SELF = "self",
+    /**
+     * Target relies on executing the target against all dependencies first.
+     *
+     * This is usually done for build like targets where you want to build all
+     * dependant projects first.
+     */
+    DEPENDENCIES = "dependencies",
+  }
+
+  /**
+   * Represents an NX Target Dependency.
+   */
+  export interface TargetDependency {
+    /**
+     * Projen target i.e: build, test, etc
+     */
+    readonly target: string;
+
+    /**
+     * Target dependencies.
+     */
+    readonly projects: TargetDependencyProject;
+  }
+
+  /**
+   * Named inputs config
+   * @see https://nx.dev/reference/nx-json#inputs-&-namedinputs
+   */
+  export interface NamedInputs {
+    readonly [name: string]: string[];
+  }
+
+  /**
+   * Target defaults config
+   * @see https://nx.dev/reference/nx-json#target-defaults
+   */
+  export interface TargetDefaults {
+    readonly [name: string]: ProjectTarget;
+  }
+
+  /**
+   * NX workspace configurations.
+   *
+   * @see https://nx.dev/configuration/packagejson
+   */
+  export interface WorkspaceConfig {
```

Comment:
Is this a breaking change given previously this was called NXConfig?

Reply:
Yes it is breaking, but only low-level config within nx-monorepo which should only be used by internals. As discussed I have updated the nx-monorepo README to include breaking details, along with commit message to denote such as well.

Given this is still v0, it is expected to break and we have now added sufficient messaging for potential external users.